### PR TITLE
WebCodecs API Overview page.

### PR DESCRIPTION
**Pull request by rachelandrew**
Adding the API overview for `WebCodecs`.

Reviewer: @jpmedley

### CODE REVIEW

**Comment by teoli2003** [comment]
Note that `GroupData.json` is now in mdn/content. I think it may be good to add the relevant entry in this very same PR (It allows to test/review the sidebars)

**Comment by rachelandrew** [comment]
@teoli2003 is it ok to list interfaces in the GroupData file that are not written yet? I'm currently working through these, there are some in PR and some not done yet. I expect to finish them next week.

If not I'll do the PR for the file with the last of the interfaces.

**Comment by teoli2003** [comment]
It is ok (the sidebar macros are pretty resilient), but if you prefer to wait, just wait :-)

File: files/en-us/web/api/webcodecs_api/index.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+{{DefaultAPISidebar("WebCodecs API")}}
+
+The **WebCodecs API** gives web developers low-level access to the individual frames of a video stream and chunks of audio.
+It is useful for web applications that require full control over the way media is processed.
+For example, video or audio editors, and video conferencing.
+
+##  Concepts and Usage
+
+Many Web APIs use media codecs internally.
+For example, the {{domxref("Web Audio API")}}, and {{domxref("WebRTC API")}}.
```
Comment (jpmedley) [inline files/en-us/web/api/webcodecs_api/index.md:19]:
```suggestion
For example, the {{domxref("Web Audio API")}}, and the {{domxref("WebRTC API")}}.
```

File: files/en-us/web/api/webcodecs_api/index.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+However these APIs do not allow developers to work with individual frames of a video stream and unmuxed chunks of encoded audio or video.
+
+Web developers have typically used WebAssembly in order to get round this limitation,
+and to work with media codecs in the browser.
+However this requires additional bandwidth to download codecs that already exist in the browser,
+reduced performance and power efficiency, and an additional development overhead.
```
Comment (jpmedley) [inline files/en-us/web/api/webcodecs_api/index.md:25]:
```suggestion
reducing performance and power efficiency, and adding additional development overhead.
```

File: files/en-us/web/api/webcodecs_api/index.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+- {{domxref("ImageDecoder")}}
+  - : Unpacks and decodes image data, giving access to the sequence of frames in an animated image.
+- {{domxref("ImageTrackList")}}
+  - : Represents the list of tracks available in the image.
+- {{domxref("ImageTrack")}}
+  - : Represents an individual image track.
+
+## Examples
+
+In the following example frames are returned from a {{domxref("MediaStreamTrackProcessor")}}, then encoded.
```
Comment (jpmedley) [inline files/en-us/web/api/webcodecs_api/index.md:59]:
```suggestion
In the following example, frames are returned from a {{domxref("MediaStreamTrackProcessor")}}, then encoded.
```

**Review by jpmedley — CHANGES_REQUESTED** [review]

**Comment by rachelandrew** [comment]
> It is ok (the sidebar macros are pretty resilient), but if you prefer to wait, just wait :-)

that's added now.

**Review by jpmedley — APPROVED** [review]

**Pull request merged**